zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/ElasticsearchRepository.java

```diff
@@ -0,0 +1,278 @@
+            .settings(
+                settings ->
+                    settings.lifecycle(lifecycle -> lifecycle.name(retention.getPolicyName())))
+            .build();
+
+    return client
+        .indices()
+        .exists(existsRequest)
+        .thenComposeAsync(
+            response -> {
+              if (!response.value()) {
+                // TODO: verify this is the expected behavior
+                return CompletableFuture.completedFuture(null);
+              }
+
+              return client.indices().putSettings(settingsRequest);
+            },
+            executor)
+        .thenApplyAsync(ok -> null, executor);
+  }
+
+  @Override
+  public CompletableFuture<Void> deleteDocuments(
+      final String sourceIndexName,
+      final String idFieldName,
+      final List<String> processInstanceKeys) {
+    final TermsQuery termsQuery = buildIdTermsQuery(idFieldName, processInstanceKeys);
+    final var request =
+        new DeleteByQueryRequest.Builder()
+            .index(sourceIndexName)
+            .waitForCompletion(true)
+            .slices(AUTO_SLICES)
+            .conflicts(Conflicts.Proceed)
+            .query(q -> q.terms(termsQuery))
+            .build();
+
+    final var timer = Timer.start();
+    return client
+        .deleteByQuery(request)
+        .whenCompleteAsync((ignored, error) -> metrics.measureArchiverDelete(timer), executor)
+        .thenApplyAsync(DeleteByQueryResponse::total, executor)
+        .thenApplyAsync(ok -> null, executor);
+  }
+
+  @Override
+  public CompletableFuture<Void> reindexDocuments(
+      final String sourceIndexName,
+      final String destinationIndexName,
+      final String idFieldName,
+      final List<String> processInstanceKeys) {
+    final var source =
+        new Source.Builder()
+            .index(sourceIndexName)
+            .query(q -> q.terms(buildIdTermsQuery(idFieldName, processInstanceKeys)))
+            .build();
+    final var request =
+        new ReindexRequest.Builder()
+            .source(source)
+            .dest(dest -> dest.index(destinationIndexName))
+            .conflicts(Conflicts.Proceed)
+            .scroll(REINDEX_SCROLL_TIMEOUT)
+            .slices(AUTO_SLICES)
+            .build();
+
+    final var timer = Timer.start();
+    return client
+        .reindex(request)
+        .whenCompleteAsync((ignored, error) -> metrics.measureArchiverReindex(timer), executor)
+        .thenApplyAsync(ignored -> null, executor);
+  }
+
+  @Override
+  public void close() throws Exception {
+    // TODO: verify if I close the transport that any pending futures are cancelled
+    client._transport().close();
+  }
+
+  private Aggregation createFinishedInstancesAggregation() {
+    final var dateAggregation =
+        AggregationBuilders.dateHistogram()
+            .field(ListViewTemplate.END_DATE)
+            .calendarInterval(rolloverInterval)
+            .format(config.getRolloverDateFormat())
```

Comment:
❓  In the original implementation here we use `getElsRolloverDateFormat`

Reply:
Oh, good catch!